<?xml version="1.0"?>
<!-- Launch file for Image Rectifier -->
<launch>
    <arg name="image_src_00" default="/port_0/camera_0/image_raw/compressed" />
    <arg name="camera_info_src_00" default="/gmsl_camera_0_0/camera_info" />
    <arg name="image_rectified_out_00" default="/image_rectified_00" />
    
    <arg name="image_src_01" default="/port_0/camera_1/image_raw/compressed" />
    <arg name="camera_info_src_01" default="/gmsl_camera_0_1/camera_info" />
    <arg name="image_rectified_out_01" default="/image_rectified_01" />
    
    <arg name="image_src_02" default="/port_0/camera_2/image_raw/compressed" />
    <arg name="camera_info_src_02" default="/gmsl_camera_0_2/camera_info" />
    <arg name="image_rectified_out_02" default="/image_rectified_02" />

	<arg name="image_src_03" default="/port_0/camera_3/image_raw/compressed" />
    <arg name="camera_info_src_03" default="/gmsl_camera_0_3/camera_info" />
    <arg name="image_rectified_out_03" default="/image_rectified_03" />

	<arg name="image_src_10" default="/port_1/camera_0/image_raw/compressed" />
    <arg name="camera_info_src_10" default="/gmsl_camera_1_0/camera_info" />
    <arg name="image_rectified_out_10" default="/image_rectified_10" />
    
    <arg name="image_src_11" default="/port_1/camera_1/image_raw/compressed" />
    <arg name="camera_info_src_11" default="/gmsl_camera_1_1/camera_info" />
    <arg name="image_rectified_out_11" default="/image_rectified_11" />
    
    <arg name="image_src_20" default="/port_2/camera_0/image_raw/compressed" />
    <arg name="camera_info_src_20" default="/gmsl_camera_2_0/camera_info" />
    <arg name="image_rectified_out_20" default="/image_rectified_20" />

	<arg name="image_src_21" default="/port_2/camera_1/image_raw/compressed" />
    <arg name="camera_info_src_21" default="/gmsl_camera_2_1/camera_info" />
    <arg name="image_rectified_out_21" default="/image_rectified_21" />
    <arg name="imshow" default="true" />
    
    <!-- autoware image_processor for plum_blob or fisheye -->
    <node pkg="image_processor" type="image_rectifier" name="image_rectifier_00" output="screen">
		<param name="image_type" value="compressed" />
        <param name="image_src" value="$(arg image_src_00)" />
        <param name="camera_info_src" value="$(arg camera_info_src_00)" />
        <param name="image_rectified" value="$(arg image_rectified_out_00)" />
        <param name="image_show" value="$(arg imshow)" type="bool"/>
    </node>
    <node pkg="image_processor" type="image_rectifier" name="image_rectifier_01" output="screen">
		<param name="image_type" value="compressed" />
        <param name="image_src" value="$(arg image_src_01)" />
        <param name="camera_info_src" value="$(arg camera_info_src_01)" />
        <param name="image_rectified" value="$(arg image_rectified_out_01)" />
		<param name="image_show" value="$(arg imshow)" type="bool"/>
    </node>
    <node pkg="image_processor" type="image_rectifier" name="image_rectifier_02" output="screen">
		<param name="image_type" value="compressed" />
        <param name="image_src" value="$(arg image_src_02)" />
        <param name="camera_info_src" value="$(arg camera_info_src_02)" />
        <param name="image_rectified" value="$(arg image_rectified_out_02)" />
        <param name="image_show" value="$(arg imshow)" type="bool"/>
    </node>
	<node pkg="image_processor" type="image_rectifier" name="image_rectifier_03" output="screen">
		<param name="image_type" value="compressed" />
        <param name="image_src" value="$(arg image_src_03)" />
        <param name="camera_info_src" value="$(arg camera_info_src_03)" />
        <param name="image_rectified" value="$(arg image_rectified_out_03)" />
        <param name="image_show" value="$(arg imshow)" type="bool"/>
    </node>
	<node pkg="image_processor" type="image_rectifier" name="image_rectifier_10" output="screen">
		<param name="image_type" value="compressed" />
        <param name="image_src" value="$(arg image_src_10)" />
        <param name="camera_info_src" value="$(arg camera_info_src_10)" />
        <param name="image_rectified" value="$(arg image_rectified_out_10)" />
        <param name="image_show" value="$(arg imshow)" type="bool"/>
    </node>
    <node pkg="image_processor" type="image_rectifier" name="image_rectifier_11" output="screen">
		<param name="image_type" value="compressed" />
        <param name="image_src" value="$(arg image_src_11)" />
        <param name="camera_info_src" value="$(arg camera_info_src_11)" />
        <param name="image_rectified" value="$(arg image_rectified_out_11)" />
        <param name="image_show" value="$(arg imshow)" type="bool"/>
    </node>
    <node pkg="image_processor" type="image_rectifier" name="image_rectifier_20" output="screen">
		<param name="image_type" value="compressed" />
        <param name="image_src" value="$(arg image_src_20)" />
        <param name="camera_info_src" value="$(arg camera_info_src_20)" />
        <param name="image_rectified" value="$(arg image_rectified_out_20)" />
        <param name="image_show" value="$(arg imshow)" type="bool"/>
    </node>
	
	<node pkg="image_processor" type="image_rectifier" name="image_rectifier_21" output="screen">
		<param name="image_type" value="compressed" />
        <param name="image_src" value="$(arg image_src_21)" />
        <param name="camera_info_src" value="$(arg camera_info_src_21)" />
        <param name="image_rectified" value="$(arg image_rectified_out_21)" />
        <param name="image_show" value="$(arg imshow)" type="bool"/>
    </node>
</launch>
